src/cases/containers/case-share-confirm/case-share-confirm.component.ts

```diff
@@ -0,0 +1,32 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { SharedCase } from '@hmcts/rpx-xui-common-lib/lib/models/case-share.model';
+import { select, Store } from '@ngrx/store';
+import { Observable } from 'rxjs';
+import * as fromCasesFeature from '../../store';
+import { LoadShareCase, LoadUserFromOrgForCase } from '../../store/actions';
+import * as fromCaseList from '../../store/reducers';
+
+@Component({
+  selector: 'exui-case-share-confirm',
+  templateUrl: './case-share-confirm.component.html',
+  styleUrls: ['./case-share-confirm.component.scss']
+})
+export class CaseShareConfirmComponent implements OnInit {
+
+  public shareCases$: Observable<SharedCase[]>;
+  public shareCases: SharedCase[];
+
+  constructor(public store: Store<fromCaseList.State>) {
+
+  }
+
+  public ngOnInit() {
+    this.shareCases$ = this.store.pipe(select(fromCasesFeature.getShareCaseListState));
+    this.shareCases$.subscribe(shareCases => {
+      this.shareCases = shareCases;
+    });
+    // call api to retrieve case assigned users
+    this.store.dispatch(new LoadShareCase(this.shareCases));
```

Comment:
it doesn't need to call the api to get the list of assigned users again, as everyting has been up to date on the store, the sharedWith[] is used on the previous page only, but on confirmation page it only needs the data from store with pendingShares[]-ToBeAdded, and pendingUnshares[]-ToBeRemoved.

Reply:
Thanks for that. Done.